### PR TITLE
boost 1.79.0

### PR DESCRIPTION
**Pull request by fxcoudert**
Previous attempt in https://github.com/Homebrew/homebrew-core/pull/99281 appears to have gotten stuck, so here we go again.

### CODE REVIEW

**Comment by fxcoudert** [comment]
It looks like `boost-build` now installs all of boost headers and cmake support files, and therefore clashes with our `boost` formula. 

Also: it looks like the project is now called `b2` (https://www.boost.org/doc/libs/1_79_0/tools/build/doc/html/index.html#_introduction) and has separate releases from boost: https://github.com/bfgroup/b2/releases Should we switch to that?

**Comment by fxcoudert** [comment]
- pcl build issue: https://github.com/PointCloudLibrary/pcl/issues/5261
- pdal and visp therefore fail to test

**Comment by SMillerDev** [comment]
If https://github.com/PointCloudLibrary/pcl/commit/bc3032e8eede7998a0b1138c6828e70f169494ac is all that's needed I'll take a stab at an upstream PR later

**Comment by SMillerDev** [comment]
There already was one it seems: https://github.com/PointCloudLibrary/pcl/commit/e964409b4accfd9070093dbc3c9cf5fb216cd877

**Comment by fxcoudert** [comment]
Not sure why the pcl patch did not take? But the upstream issue hinted that the latest cmake would deal with the issue anyway (3.23.2), so I'm rebasing and trying to build again.

**Comment by fxcoudert** [comment]
Fixing `pcl` build by passing `-DBoost_USE_DEBUG_RUNTIME:BOOL=OFF` so that it accepts our boost_filesystem.

**Comment by SMillerDev** [comment]
Gonna need a revision bump for `c10t` too but otherwise it looks good on ARM

**Comment by SMillerDev** [comment]
I guess the self-hosted runner is tied up?

**Comment by fxcoudert** [comment]
Linux needs a lot more revision bumps…

**Comment by cho-m** [comment]
ARM and Linux failures:
```
Error: 6 failed steps!
brew install --verbose --build-bottle cryptominisat
brew install --verbose --build-bottle rdkit
brew install --verbose --build-bottle simgrid
brew install --build-from-source cryptominisat
brew install --only-dependencies --verbose --build-bottle stp
brew install --verbose --build-bottle stp
```

**Comment by iMichka** [comment]
Rebase. I also rewrote 51 commit messages which mentioned `gcc` instead of `boost`.

**Comment by SMillerDev** [comment]
Standardese seems to be missing a revision bump, but we can do that in a new PR too.

**Review by SMillerDev — DISMISSED** [review]

**Comment by cho-m** [comment]
Linux:
```
Error: 6 failed steps!
brew linkage --test cryfs
brew test --retry --verbose cryfs
brew linkage --test fastnetmon
brew test --retry --verbose fastnetmon
brew linkage --test standardese
brew test --retry --verbose standardese
```

10.15 will need a re-run.

11 only hit one reported failure during a fetch, but I don't think that impacting bottle build.

**Comment by iMichka** [comment]
`fastnetmon`, `standardese` and `cryfs` needed a revision bump.
As 10.15 failed completely, I added the 3 changes and will do a full build again.

**Comment by cho-m** [comment]
May want to hold off until #104742 is merged as `etcd-cpp-apiv3` will conflict.

File: Formula/etcd-cpp-apiv3.rb
Comment (cho-m) [inline Formula/etcd-cpp-apiv3.rb:7]: Needs rebase with new `revision 3` -- https://github.com/Homebrew/homebrew-core/commit/1e3494f22beb613d36c07d6b13a5b388f076d94f

**Comment by SMillerDev** [comment]
Merged now, @iMichka got some time to rebase this?

**Comment by iMichka** [comment]
Yes will do

**Comment by carlocab** [comment]
Build failures on Linux:

- `collada-dom` -- wrong C++ standard?
- `metaproxy` -- can't find `libxml` headers

Also a couple of linkage failures.

**Comment by cho-m** [comment]
> - `collada-dom` -- wrong C++ standard?

Maybe from an icu4c update? At least I think it is failing on those headers. Was going to try to repro while removing some bundled deps in #104926

> - `metaproxy` -- can't find `libxml` headers

Should be due to yaz. Trying out workaround in #104925 (until we have better way to handle all formulae like we were discussing in qt/qt@5)

---

Not sure if we want to fix issues afterward (temp lose some Linux bottles) or rerun (~1 day?)

**Comment by carlocab** [comment]
> Not sure if we want to fix issues afterward (temp lose some Linux bottles) or rerun (~1 day?)

Let's fix them afterward. All the broken formulae on Linux (`collada-dom`, `metaproxy`, `fastnetmon`, `quick-lint-js`) have very few installs on Linux. I don't think it's worth delaying this bump further (and slamming CI for another round).

**Review by carlocab — DISMISSED** [review]
Let's merge this and fix Linux failures after.

Waiting longer delays this much delayed update further, and risks merge conflicts.

**Review by iMichka — DISMISSED** [review]

**Comment by cho-m** [comment]
Merge conflict due to `cryptominisat` modification in #104288

**Review by carlocab — DISMISSED** [review]
Rebased.

**Comment by cho-m** [comment]
On `collada-dom`, it looks like this is side effect of adding `icu4c` dependency to `libxml2` (#103721). Another PR (#104974) hit a similar failure for `libvirt-glib` test, so this may be seen in other formulae. Looks like the `icu4c` headers will need `-std=c++11` when compiling with GCC-5. I can rebase #104926 once this PR is merged.

**Comment by fxcoudert** [comment]
```
cp: error writing '/tmp/10157720220703-1887-r0rem1/freeling--4.2_7.arm64_big_sur.bottle.tar.gz': No space left on device
```

**Comment by SMillerDev** [comment]
Pulling it locally now

**Comment by carlocab** [comment]
Don't we have a way of using the self-hosted runner to publish PRs where GitHub runners don't have enough space? I don't know how to use it though. @Bo98?

**Comment by SMillerDev** [comment]
Me neither, but I made it in the end

**Comment by cho-m** [comment]
Dropped Linux bottles:
- [ ] `collada-dom` -- should be available with #104926
- [ ] `metaproxy` -- fixed base `yaz` issue (#104925), but also need to fix propagated issue in `yazpp` (#105023). Once done, hopefully should be able to dispatch Linux bottle job for `metaproxy`.
- [ ] `fastnetmon`
- [ ] `quick-lint-js`

**Pull request closed without merging**
